hypertrace-gradle-publish-maven-central-plugin/src/main/java/org/hypertrace/gradle/publishing/PublishMavenCentralPlugin.java

```diff
@@ -199,4 +199,21 @@ private void validateExtensionAtConfigurationTime() {
     }
   }
 
+  private void validateGradlePropertiesBeforePublishTask() {
+    project.getTasks().withType(PublishToMavenRepository.class).stream().forEach(task -> {
```

Comment:
updated the check on `publish` task. It will fail with following error message:
```
> Task :filter-api:publish FAILED
Caching disabled for task ':filter-api:publish' because:
  Build cache is disabled
Task ':filter-api:publish' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
:filter-api:publish (Thread[Execution worker for ':',5,main]) completed. Took 0.001 secs.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':filter-api:publish'.
> Missing expected gradle property: ossrhUsername. It should be added in your ~/.gradle/gradle.properties file, or in a an environment variable of the form ORG_GRADLE_PROJECT_ossrhUsername
```

Task `publishToMavenLocal` is successful even if `ossrhUsername` and `ossrhPassword` properties are not defined.